### PR TITLE
Compile and check private_generics during the execution of rooch move test

### PR DESCRIPTION
**Pull request by steelgeek091**
When executing the command `rooch move test`, check the functions decorated with the `#[test]` attribute to ensure they comply with the rules for calling `private_generics`.

resolve https://github.com/rooch-network/rooch/issues/335.

### CODE REVIEW

File: crates/rooch/src/commands/move_cli/commands/unit_test.rs [hunk 1]
```diff
@@ -36,6 +43,45 @@ impl Test {
...
+        let _ = run_extended_checks(&global_env);
+
+        if global_env.diag_count(Severity::Warning) > 0 {
+            let mut buffer = Buffer::ansi();
+            global_env.report_diag(&mut buffer, Severity::Warning);
+            let buffer_output = String::from_utf8_lossy(buffer.as_slice()).to_string();
+            let re = Regex::new("(/.*)(.move:[0-9]+:[0-9]+)").unwrap();
+            let ret = re
+                .replace(buffer_output.as_str(), "/tmp/tempfile$2".to_string())
+                .to_string();
```
Comment (jolestar) [inline crates/rooch/src/commands/move_cli/commands/unit_test.rs:78]: do not need to replace output file names in unit tests output.

File: moveos/moveos-verifier/src/build.rs [hunk 2]
```diff
@@ -51,6 +269,34 @@ pub fn build_model(
     )
 }
 
+pub fn build_model_internal(
```
Comment (jolestar) [inline moveos/moveos-verifier/src/build.rs:272]: Why `internal`? it is a public fn.

File: crates/rooch/src/commands/move_cli/commands/unit_test.rs (hunk 1 above)
Comment (steelgeek091) [reply crates/rooch/src/commands/move_cli/commands/unit_test.rs]: That's correct, here it has been changed to not replace the content.

File: moveos/moveos-verifier/src/build.rs (hunk 2 above)
Comment (steelgeek091) [reply moveos/moveos-verifier/src/build.rs]: Makes sense, the function name has been changed to `build_model_with_test_attr`.

**Review by jolestar — APPROVED** [review]

**Review by baichuan3 — APPROVED** [review]

**Pull request merged**